scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -1875,7 +1889,7 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
     }).map(_.reduceWith(toParamClause(None)))
 
   def implicitClosure(location: Location): Term.Function = {
-    require(token.isNot[KwImplicit] && debug(token))
+    assert(token.isNot[KwImplicit], s"Expected implicit keyword but found ${token.name}")
```

Comment:
actually, it's the opposite. expected to be _after_ the implicit keyword... why not make implicitClosure private and remove the assert?

Reply:
Done! That's much better, I don't love putting asserts anywhere in the code aside from tests.